modules/virt-configuring-vm-real-time.adoc

```diff
@@ -0,0 +1,94 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/advanced_vm_management/virt-configuring-cluster-realtime-workloads.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="virt-configuring-vm-real-time_{context}"]
+= Configuring a virtual machine for real-time workloads
+
+You can run real-time workloads on virtual machines (VMs).
+
+.Prerequisites
+* Your cluster is configured to run real-time workloads.
+
+.Procedure
+. Edit the `VirtualMachine` manifest to include information about CPU topology, CRI-O annotations, and huge pages:
++
+.Example `VirtualMachine` manifest
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  annotations:
+    cpu-load-balancing.crio.io: disable <1>
+    cpu-quota.crio.io: disable <2>
+    irq-load-balancing.crio.io: disable <3>
+  name: realtime-vm
+spec:
+  domain:
+    cpu:
+      dedicatedCpuPlacement: true
+      isolateEmulatorThread: true
+      model: host-passthrough
+      numa:
+        guestMappingPassthrough: {}
+      realtime: {}
+      sockets: 1 <4>
+      cores: 4 <5>
+      threads: 1
+    devices:
+      autoattachGraphicsDevice: false
+      autoattachMemBalloon: false
+      autoattachSerialConsole: true
+    ioThreadsPolicy: auto
+    memory:
+      guest: 4Gi
+      hugepages:
+        pageSize: 1Gi <6>
+  terminationGracePeriodSeconds: 0
+# ...
+----
+<1> This annotation specifies that load balancing is disabled for CPUs that are used by the container.
+<2> This annotation specifies that the CPU quota is disabled for CPUs that are used by the container.
+<3> This annotation specifies that Interrupt Request (IRQ) load balancing is disabled for CPUs that are used by the container.
+<4> The number of sockets inside the VM. This field must be set to `1` for the CPUs to be scheduled from the same Non-Uniform Memory Access (NUMA) node.
+<5> The number of cores inside the VM. This must be a value greater than or equal to `1`. In this example, the VM is scheduled with 5 hyper-threads or 10 CPUs.
+<6> The size of the huge pages. The possible values for x86-64 architecture are 1Gi and 2Mi. In this example, the request is for 4 huge pages of size 1Gi.
+
+. Save and exit the editor.
+. Apply the `VirtualMachine` manifest:
++
+[source,terminal]
+----
+$ oc apply -f <file_name>.yaml
+----
+
+. Configure the guest operating system. The following example shows the configuration steps for {op-system-base} 8 OS:
```

Comment:
If this should be set on the guest VM, then the user should first be instructed to access the VM (`virtctl console <vm>`).
This also means adding the installation of `virtctl` as a prerequisite.